OAuth/ResourceOwner/AbstractResourceOwner.php

```diff
@@ -231,17 +231,13 @@ protected function httpRequest($url, $content = null, $headers = array(), $metho
         $request  = new HttpRequest($method, $url);
         $response = new HttpResponse();
 
-        $contentLength = 0;
         if (is_string($content)) {
-            $contentLength = strlen($content);
-        } elseif (is_array($content)) {
-            $contentLength = strlen(implode('', $content));
+            $headers =  array_merge(array('Content-Length' => strlen($content)), $headers);
```

Comment:
Please notice that this is a wrong check because of this:

> If you need length of string in bytes (**strlen cannot be trusted anymore because of mbstring.func_overload**) you should use `<?php mb_strlen($string, '8bit'); ?>`.
> – http://fr2.php.net/manual/en/function.mb-strlen.php

I keept the old way of doing because:

1. Most part of the time people don't hate developers enough to activate this option
2. It was done like this before